apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/service/config/ConfigServiceWithCache.java

```diff
@@ -149,6 +161,39 @@ public void handleMessage(ReleaseMessage message, String channel) {
     }
   }
 
+  @Override
+  public ImmutableMap<String, Release> findReleasesByReleaseKeys(Set<String> releaseKeys) {
+    try {
+      ImmutableMap<String, Long> releaseKeyMap = releaseKeyCache.getAll(releaseKeys);
+      if (CollectionUtils.isEmpty(releaseKeyMap)) {
+        return null;
+      }
+
+      Map<Long, Optional<Release>> releasesMap = configIdCache.getAll(releaseKeyMap.values());
+      if (CollectionUtils.isEmpty(releasesMap)) {
+        return null;
+      }
+
+      Map<String, Release> releases = new HashMap<>();
+      releaseKeyMap.forEach((releaseKey,id) -> {
+        if (releasesMap.containsKey(id)) {
+          Release release = releasesMap.get(id).orElse(null);
+          if(release!=null){
+            releases.put(releaseKey, releasesMap.get(id).orElse(null));
+          }
+        }
+      });
+
+      if(releases.size()>0){
+        return ImmutableMap.copyOf(releases);
+      }
+
+    } catch (ExecutionException e) {
```

Comment:
Would it be better to return the exception to the caller instead of leaving it out?